app/screens/permalink/permalink.tsx

```diff
@@ -135,20 +144,113 @@ function Permalink({channel, isCRTEnabled, postId}: Props) {
 
     useEffect(() => {
         (async () => {
-            if (channel?.id) {
-                const data = await fetchPostsAround(serverUrl, channel.id, postId, 5, isCRTEnabled);
+            if (channelId) {
+                const data = await fetchPostsAround(serverUrl, channelId, postId, 5, isCRTEnabled);
+                if (data.error) {
+                    setError({unreachable: true});
+                }
                 if (data?.posts) {
-                    setLoading(false);
                     setPosts(data.posts);
                 }
+                setLoading(false);
+                return;
+            }
+
+            const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+            if (!database) {
+                setError({unreachable: true});
+                setLoading(false);
+                return;
+            }
+
+            // If a team is provided, try to join the team, but do not fail here, to take into account:
+            // - Wrong team name
+            // - DMs/GMs
+            let joinedTeam: Team | undefined;
+            if (teamName && !isTeamMember) {
+                const fetchData = await fetchTeamByName(serverUrl, teamName, true);
+                joinedTeam = fetchData.team;
+
+                if (joinedTeam) {
+                    const addData = await addUserToTeam(serverUrl, joinedTeam.id, currentUserId);
+                    if (addData.error) {
+                        joinedTeam = undefined;
+                    }
+                }
+            }
+
+            const {post} = await fetchPostById(serverUrl, postId, true);
```

Comment:
Can you fetch the post without joining the channel? I can remember if this is handled by fetchTeamByName as in fetching channels

Reply:
Apparently you can, and it is the way it works on webapp and V1. Not sure why is it that way, but if the channel is public, you can.

Reply:
Oh wow, that would make channel preview without joining so much easier to implement.

Reply:
Unless is an admin only thing?

Reply:
It is an all users thing. If not, there is no way to know from which channel is a post just by the id, and then join the channel :sweat_smile:
The admin only thing is that you can do the same... with private channels. Not sure about dms/gms, but I think that is already controlled in the server.